packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/search/enhanced/cross_document_intelligence.py

```
@@ -1366,43 +1773,421 @@ def _get_complementary_content_type_score(self, target_doc: SearchResult, candid
+                            ids=[doc_id]
+                        ),
+                        timeout=5.0  # 5 second timeout per document
+                    )
+                    if result and len(result) > 0:
+                        # Extract dense vector embedding
+                        vector = result[0].vector
+                        if isinstance(vector, dict) and "dense" in vector:
+                            embeddings[doc_id] = vector["dense"]
+                        elif isinstance(vector, list):
+                            embeddings[doc_id] = vector
+                except asyncio.TimeoutError:
```

Ruff (0.12.2)

Replace aliased errors with `TimeoutError`

Replace `asyncio.TimeoutError` with builtin `TimeoutError`

(UP041)